iFixit/react-commerce
# Refactor project structure and change product list path

Closes #118 

## Summary

This PR is a major project refactor to fix inconsistencies in the codebase and define clear rules to scale the project and make the codebase easier to understand.

### Context

To contextualize the changes, here are some of the issues that are addressed:

- When it was decided to rename from "collections" to "product-list", many files were involved and initially I opted for doing only the minimal changes required to account for changes in the Strapi API (I didn't want to open a big PR). Turns out that gradually migrating all the files is not really working out and seing both terms still being used might confuse people working on the project down the road
- While doing work for #158 I've realized that some of the business logic related to data fetching is not well defined as types were completely inferred by Strapi. This means that breaking changes from Strapi API tends to propagate through the code base. Also in the future we may want to change the API underneath certain use cases (e.g. using iFixit API).
- Views were spread between `/features` `/pages` and `/components` and it wasn't always clear where each component belong

I took this opportunity to review the project structure overall and define a better structure which vaguely resembles an MVC approach (inspired by [Vercel Commerce](https://github.com/vercel/commerce)):

- `pages`: contains the Next.js app pages (think of these like routes/controllers)
- `models`: contains business logic (e.g. how to fetch product list from API, how to subscribe to newsletter, etc.)
- `components`: contains the React view components
- `helpers`: contains reusable custom app-related helper functions
- `lib`: contains custom libraries that can stand on their own (e.g. `lib/algolia`). Think of these as packages that potentially could be used in other projects.
- `assets`: contains assets imported from view components (e.g. svg illustrations)
- `public`: contains the static files
- `config`: contains app configurable settings (e.g. environment variables, constants, etc.)

## Changelog

- Change product lists path from `/collections` ➡️  `/store`
- Refactor `/frontend` project folder

## QA

1. Visit the Vercel PR preview
2. Verify that product lists are visible under `/store` (e.g. `/store/parts`)
3. Verify that everything else works like in the main branch (https://react-commerce.vercel.app/)

> Access to Strapi deployment: https://refactor-project-structure.govinor.com/admin
> User: `strapi@ifixit.com`
> Pwd: `Password1`
